### PR TITLE
Merge release-2.6-updates to main

### PR DESCRIPTION
**Pull request by 56quarters**
In this PR I'm merging the release-2.6-updates branch to main branch. Please merge this PR using a **merge commit** or by using `tools/release/merge-approved-pr-branch-to-main.sh` script.

### CODE REVIEW

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-2.5.0
\ No newline at end of file
+2.6.0-rc.0
```
Comment (pracucci) [inline VERSION:1]: Uhm... this should be 2.6.0.

Comment (pracucci) [reply VERSION:1]: Have we released 2.6.0 stable with the -rc.0 version inside it?

Comment (56quarters) [reply VERSION:1]: No, this seems like a problem with just this PR: https://github.com/grafana/mimir/blob/release-2.6/VERSION

Comment (56quarters) [reply VERSION:1]:
```
$ ./mimir-linux-amd64 -version
Mimir, version 2.6.0 (branch: release-2.6, revision: 27698f399)
  go version:       go1.19.3
  platform:         linux/amd64
```

**Pull request closed without merging**